test/__snapshot__/parser/module/top-level-await/inside-arrow.md

```diff
@@ -0,0 +1,109 @@
+# Kataw parser test case
+
+## Options
+
+`````js
+{ jsx: true, disableWebCompat: true, module: true }
+`````
+
+## Input
+
+`````js
+() => await 0;
+`````
+
+## Output
+
+### CST
+
+```javascript
+{
+    "kind": 122,
+    "directives": [],
+    "statements": [
+        {
+            "kind": 120,
+            "expression": {
+                "kind": 271,
+                "arrowToken": {
+                    "kind": 10,
+                    "flags": 0,
+                    "start": 2,
+                    "end": 5
+                },
+                "typeParameters": null,
+                "parameters": [],
+                "asyncKeyword": null,
+                "returnType": null,
+                "contents": {
+                    "kind": 216,
+                    "functionStatementList": {
+                        "kind": 217,
+                        "directives": [],
+                        "statements": [
+                            {
+                                "kind": 120,
+                                "expression": {
+                                    "kind": 208,
+                                    "awaitKeyword": {
+                                        "kind": 90224,
+                                        "flags": 0,
+                                        "start": 5,
+                                        "end": 11
+                                    },
+                                    "expression": {
+                                        "kind": 201392130,
+                                        "text": 0,
+                                        "rawText": "0",
+                                        "flags": 96,
+                                        "start": 11,
+                                        "end": 13
+                                    },
+                                    "flags": 32,
+                                    "start": 5,
+                                    "end": 13
+                                },
+                                "flags": 16,
+                                "start": 5,
+                                "end": 14
+                            }
+                        ],
+                        "flags": 32,
+                        "start": 5,
+                        "end": 14
+                    },
+                    "flags": 32,
+                    "start": 5,
+                    "end": 14
+                },
+                "flags": 32,
+                "start": 0,
+                "end": 14
+            },
+            "flags": 16,
+            "start": 0,
+            "end": 14
+        }
+    ],
+    "isModule": true,
+    "source": "() => await 0;",
+    "fileName": "__root__",
+    "flags": 0,
+    "start": 0,
+    "end": 14
+}
+```
+
+### Printed
+
+```javascript
+
+```
+
+### Diagnostics
+
+```javascript
+✖ The parser expected to find a '}' to match the '{' token here - start: 14, end: 14
```

Comment:
nits: there is a room to improve the message, e.g. "expect 'await' in an async function."

Reply:
I agree. It got caught in error recovery mode, and made some other tests fail. I have fixed it now.